apps/language_server/lib/language_server.ex

```diff
@@ -143,8 +143,9 @@ defmodule ElixirLS.LanguageServer do
     enum_ex_path = Enum.module_info()[:compile][:source]
 
     elixir_sources_available =
-      unless File.exists?(enum_ex_path, [:raw]) do
-        dir = Path.join(enum_ex_path, "../../../..") |> Path.expand()
+      # handle possible nil from deterministic build of elixir
+      unless enum_ex_path && File.exists?(enum_ex_path, [:raw]) do
+        dir = Path.join(enum_ex_path || ~c"/", "../../../..") |> Path.expand()
```

Comment:
This seems wrong? If enum_ex_path is nil, this will return... "/../../.."?

Reply:
The `Path.expand()` will collapse it down to `/`, so the log message shows `"Elixir sources not found (checking in /). Code navigation to Elixir modules disabled."`

Main reasoning for this was to avoid changing the behavior for the output logging for the case of elixir deterministic build where source has been stripped.

This entire block is fragile, as the previous logged output would have been `/home/abuild/rpmbuild/BUILD/elixir-1.18.4-build/elixir-1.18.4` under openSUSE Tumbleweed. Which would have been equally strange, since this is an artifact from the build system, and not a path you'd ever see on a user's system.

Reply:
Got it.

Reply:
Thanks. I was confused by it when I read it :D